File: src/FASTCAMPUS/ALGO/D01/Main.ts
class myQueue<T> {
  queue: T[];
  constructor(){
    this.queue = [];
  }
  enqueue(value:T){
    this.queue.push(value);
  }

  dequeue():T|null|undefined{
    if(this.isEmpty()){
      return null;
    } else {
      return this.queue.shift();
    }
  }

  isEmpty():boolean{
    return this.queue.length === 0;
  }
}

class myStack<T>{
  stack:T[];
  constructor(){
    this.stack = [];
  }
  push(value:T):T{
    this.stack.push(value);
    return value;
  };
  pop():T|undefined{
    return this.stack.pop();
  };
  isEmpty():boolean{
    return this.stack.length === 0;
  }
}

const stack = new myStack<number>();

stack.push(1);
console.log(stack.pop());
console.log(stack.isEmpty());

File: src/FASTCAMPUS/ALGO/D01/Main.java
package FASTCAMPUS.ALGO.D01;
import java.util.ArrayList;

class Main {
  public static void main(String[] args) {
    customStack<Integer> stack = new customStack <Integer>();
    System.out.println(stack.push(1));
    System.out.println(stack.pop());
    System.out.println(stack.isEmpty());
  }
}

class customQueue<T> {
  private ArrayList<T> queue;
  customQueue(){
    queue = new ArrayList<T>(); 
  }
  public void enqueue(T value){
    queue.add(value);
  }
  public T dequeue(){
    if(queue.isEmpty()){
      return null;
    }
    return queue.remove(0);
  }

  public boolean isEmpty() {
    return queue.isEmpty();
  }

} 

class customStack <T> {
  private ArrayList<T> stack;
  customStack(){
    this.stack = new ArrayList<T>();
  }

  public T push(T value){
    if(this.stack.add(value)){
      return value;
    }
    return null;
  }

  public T pop(){
    if(this.isEmpty()){
      return null;
    } 
    return this.stack.remove(this.stack.size() - 1 );
  }

  public boolean isEmpty(){
    return this.stack.isEmpty();
  }

}
